code/modules/organs/organ_icon.dm

```diff
@@ -80,23 +65,40 @@ var/global/list/limb_icon_cache = list()
 		mob_icon.Blend(mark_s, ICON_OVERLAY) //So when it's on your body, it has icons
 		icon_cache_key += "[M][markings[M]["color"]]"
 
-	if(owner.f_style)
-		var/datum/sprite_accessory/facial_hair_style = facial_hair_styles_list[owner.f_style]
-		if(facial_hair_style && facial_hair_style.species_allowed && (species.get_bodytype(owner) in facial_hair_style.species_allowed))
-			var/icon/facial_s = new/icon("icon" = facial_hair_style.icon, "icon_state" = "[facial_hair_style.icon_state]_s")
-			if(facial_hair_style.do_colouration)
-				facial_s.Blend(rgb(owner.r_facial, owner.g_facial, owner.b_facial), ICON_ADD)
-			overlays |= facial_s
-
-	if(owner.h_style && !(owner.head && (owner.head.flags_inv & BLOCKHEADHAIR)))
-		var/datum/sprite_accessory/hair/hair_style = hair_styles_list[owner.h_style]
-		if(hair_style && (species.get_bodytype(owner) in hair_style.species_allowed))
-			var/icon/hair_s = new/icon("icon" = hair_style.icon, "icon_state" = "[hair_style.icon_state]_s")
-			var/icon/hair_s_add = new/icon("icon" = hair_style.icon_add, "icon_state" = "[hair_style.icon_state]_s")
-			if(hair_style.do_colouration && islist(h_col) && h_col.len >= 3)
-				hair_s.Blend(rgb(h_col[1], h_col[2], h_col[3]), ICON_MULTIPLY)
-				hair_s.Blend(hair_s_add, ICON_ADD)
-			overlays |= hair_s
+	if(severed) //Properly applied on intact mobs at update_icons_body
+		if(owner.should_have_organ(O_EYES))
+			var/obj/item/organ/internal/eyes/eyes = owner.internal_organs_by_name[O_EYES]
+			if(eye_icon)
+				var/icon/eyes_icon = new/icon(eye_icon_location, eye_icon)
+				if(eyes)
+					eyes_icon.Blend(rgb(eyes.eye_colour[1], eyes.eye_colour[2], eyes.eye_colour[3]), ICON_ADD)
+				else
+					eyes_icon.Blend(rgb(128,0,0), ICON_ADD)
+				mob_icon.Blend(eyes_icon, ICON_OVERLAY)
+				overlays |= eyes_icon
+
+		if(owner.lip_style && (species && (species.appearance_flags & HAS_LIPS)))
+			var/icon/lip_icon = new/icon('icons/mob/human_face.dmi', "lips_[owner.lip_style]_s")
+			overlays |= lip_icon
+			mob_icon.Blend(lip_icon, ICON_OVERLAY)
```

Comment:
Wait, this part is important though. Only the overlays part can be skipped. The mob_icon blends have to be done otherwise they won't have eyes on their body.

Reply:
Tested another sort of approach and founf out the markings are important too. Wouldn't show up in preview at all when skipped.

Reply:
Just do
```
if(severed)
    overlays |= thing
```
Is what I was intending to suggest. The mob_icon stuff needs to stay where it was.